lib/incoming_mail/handler.rb

```diff
@@ -72,7 +72,11 @@ def create_comment(msg)
       proposal = parsed_email.proposal
       user = parsed_email.comment_user
 
-      return unless user  # cannot create comment for non-existent user
+      # cannot create comment for non-existent user, or
+      # for user who is not already a subscriber
+      unless user && proposal.has_subscriber?(user)
```

Comment:
feels like the logic in this `create_comment` method is getting pretty complicated (~30 lines long). Any ideas on how we could simplify it?